---
title: "notes_code_project"
format: html
editor: visual
---

## Load data

## Tidy data:)

### Long format for smooth data anlysis

From the very small bit of visualisation I wanted to do, I could see that the replicates should actually be in 1 column (points to Magda). The reason being, that when ggplot calls for data input, it will recognize the variable as the column. *I cannot plot the measured intensities, because they have seperate x values*.

### Small things to remember

-   Remember to write cleaned/augmented files to .tsv.gz

-   remember quatation signs "" when loading libraries

##### 

## 

## Data analysis

Making functions instead of repetivie code for the volcanoplot inputs...

### Ideal plots for proteomics

#### Lasse

#### Volcano plot: Lasse + Antonio + Magda

Issues when doing the volcano plot. We want to create one column with fold of two cell-type intensities and one column with p-values of two cell-type intensities, always grouped per gene/protein.

The issue with this is how to create new variables for fold / pvalue when we are in long format? We have made a solution that works with mean + sd values only and transforming back to wide format.

Calculate the fold change of the log mean values between samples of the defined pairwise.

Calculate the false disocvery rate for proteomics data? Found in the paper?

-   FDR = FP / (FP + TP)

-   FP = false positives

-   TP = True positives

-   [false discovery rate guide](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1716-1)

The volcano plot is plotted as the log10(false_discovery rate) against on the y axis against the log2(fold change)

Simpler method is to plot the p-value instead of the log10(FDR) and we can use a cutoff value of 1% or even different ones.

-   

-   Showcase that the data is gausian/normal distributed? - prove nice data

Proteome coverage of the analyzed populations. A bar plot of the population counting the average protein group number.

Column diagrams or densitograms for the expression of the cell

## Presentation:

-   It seems impossible to make the images go directly from the "results" folder into the presentation.qmd...Right now a workable solution is done but it is not what Leon wants.

## Already dealt with / Bin

### Naming conventions of strings and column names

**Magda:** Throughout the course we mostly used the naming convention snake_case (meaning lowercase letters separated by underscore). I suggest we change all of our variable names to snake_case.

### Name redundancy in protein and gene columns

**Magda:** Some cells in the columns protein and genes contain more than one name separated by a semicolon ;. We discussed it w Arne, and he says it is 'biologically' correct to have multiple names because we can't in reality know which protein it is, but for the purpose of tidy data I think we should shorten those names to only contain one name.
