---
title: "05_analysis"
format: html
editor: visual
---

## Library

```{r}
library("tidyverse")
library("broom")
library("ggthemes")
library("ggrepel")
library("ggplot2")
library("patchwork")
```

## Data

For this section of the report, the data frame with the sample-wise imputed missing values in wide format are used.

```{r}
df_no_na_long <- read_tsv(file = here("data/03_dat_aug_long_1.tsv"))
source(here("R/99_functions.R"))
```

## Data Analysis

## CLP relative to HSC

## CMP relative to HSC

First of all we want to subset the data so we only work with two cell types. In this case hsc and gmp as they appear to be the most different.

```{r}
# Histogram that shows distribution of the fold
ggplot(df_mean_intensitities, aes(x=fold))+
  geom_histogram()+
  scale_x_continuous(limits = c(0.75, 1.25))
```

## Volcano plots

Genes with statistically significant differential expression lie above a horizontal threshold. Closer to zero indicates less change while moving away from zero in either direction indicates more change. Volcano plots provide an effective means for visualizing the direction, magnitude, and significance of changes in gene expression.Â 

```{r}
test_plot <- df_mean_intensitities |> 
pivot_wider(names_from = cell_type, values_from = c(mean, sd)) |> 
  select(mean_gmp, mean_hsc, sd_cmp, sd_hsc) |> 
  mutate(log2fold = log2(mean_cmp/mean_hsc)) |> 
  #using the home-made function pval
  mutate(pval = pval(mu1 = mean_cmp, mu2 = mean_hsc, n1 = 4, n2 = 4, s1 = sd_cmp, s2 = sd_hsc)) |> 
  select(protein_groups, log2fold, pval) |> 
  mutate(is_signif = case_when(pval <= 0.05 & log2fold > 0 ~ "up",
                               pval <= 0.05 & log2fold < 0 ~"down",
                               pval > 0.05 ~"not significant"
  ))


```

q

We will look at the development from HSC to CLP and from HSC to CMP.

And then from CMP to MEP and from CMP to GMP. Because that is the order in which the cells develop.

```{r}
library(ggrepel) 
test_plot |>
  #When are points significant? In this case if fold > 1 it is overexpressed
 # mutate(label = case_when(fold > 1 ~ protein_groups, fold < 1 ~ "")) |>
  ggplot(aes(x = log2fold, y = desc(log10(pval)), color = is_signif))+
  geom_point(size=1, alpha = 0.5)+
#  geom_text_repel(aes(label = label), size = 3)+
  theme_minimal()+
  labs(title = "Expression of CLP relative to HSC",
       x = "fold",
       y = "-log10(p)",
       caption = "proteomics data")+
  theme(legend.position = "none")+
  #The horizontal line that indicates significance - remember to check if it correct with qvalue and all
#  geom_hline(yintercept = 2)+
  scale_color_manual(values = c("overexpressed" = "#D55E00", "underexpressed" = "#0072B2", "not_significant" = "gray"))

#The threshold should be p value of 0.05
#Use q-value instead of pvalue 
#Sofias code for defining the color
#scale_color_manual(values = c("Down-regulated" = "blue", "Up-regulated" = "red", "NA" = "gray"))
```

```{r}
hsc_clp_plot <- df_volcano_clp_hsc |>
  ggplot(aes(x = fold_log2, y = desc(log10(q_val)), color = expression))+
  geom_point(size=0.5, alpha = 0.5)+
  theme_classic()+
  labs(title = "Expression of CLP relative to HSC",
       x = "Log2fold-change",
       y = "-log10(q)",
       caption = "proteomics data")+
  theme(legend.position = "none")+
  #The horizontal line that indicates significance - remember to check if it correct with qvalue and all
#  geom_hline(yintercept = 2)+
  scale_color_manual(values = c("overexpressed" = "#D55E00", "underexpressed" = "#0072B2", "not_significant" = "gray"))
```

```{r}
hsc_cmp_plot <- df_volcano_cmp_hsc |>
  ggplot(aes(x = fold_log2, y = desc(log10(q_val)), color = expression))+
  geom_point(size=0.5, alpha = 0.5)+
  theme_classic()+
  labs(title = "Expression of CMP relative to HSC",
       x = "Log2fold-change",
       y = "-log10(q)",
       caption = "proteomics data")+
  theme(legend.position = "none")+
  #The horizontal line that indicates significance - remember to check if it correct with qvalue and all
#  geom_hline(yintercept = 2)+
  #Colors are from the colorblind palette
  scale_color_manual(values = c("overexpressed" = "#D55E00", "underexpressed" = "#0072B2", "not_significant" = "gray"))
```

```{r}
cmp_mep_plot <- df_volcano_mep_cmp |>
  ggplot(aes(x = fold_log2, y = desc(log10(q_val)), color = expression))+
  geom_point(size=0.5, alpha = 0.5)+
  theme_classic()+
  labs(title = "Expression of MEP relative to CMP",
       x = "Log2fold-change",
       y = "-log10(q)",
       caption = "proteomics data")+
  theme(legend.position = "none")+
  #The horizontal line that indicates significance - remember to check if it correct with qvalue and all
#  geom_hline(yintercept = 2)+
scale_color_manual(values = c("overexpressed" = "#D55E00", "underexpressed" = "#0072B2", "not_significant" = "gray"))
```

```{r}
cmp_gmp_plot <- df_volcano_gmp_cmp |>
  ggplot(aes(x = fold_log2, y = desc(log10(q_val)), color = expression))+
  geom_point(size=0.5, alpha = 0.5)+
  theme_classic()+
  labs(title = "Expression of GMP relative to CMP",
       x = "Log2fold-change",
       y = "-log10(q)",
       caption = "proteomics data")+
  theme()+
  #The horizontal line that indicates significance - remember to check if it correct with qvalue and all
#  geom_hline(yintercept = 2)+
  scale_color_manual(values = c("overexpressed" = "#D55E00", "underexpressed" = "#0072B2", "not_significant" = "gray"))

```

```{r}
#Due to an error occuring when saving ggplot directly in the pipeline of the latest plot which apparently is caused by a recent edit in ggplot2-3.3.4 the saving of plots is done separately.
ggsave(plot = hsc_clp_plot, filename = "hsc_clp_volcano_plot.png", width = 20, height = 10, units = "cm", path = here("results"))
ggsave(plot = hsc_cmp_plot, filename = "hsc_cmp_volcano_plot.png", width = 20, height = 10, units = "cm", path = here("results"))
ggsave(plot = cmp_mep_plot, filename = "cmp_mep_volcano_plot.png", width = 20, height = 10, units = "cm", path = here("results"))
ggsave(plot = cmp_gmp_plot, filename = "cmp_gmp_volcano_plot.png", width = 20, height = 10, units = "cm", path = here("results"))
```

A future direction for the project would be to assign all the protein groups categories based on their functionality. It would for instance be interesting to see which are the proteins that are overexpressed in the different cell types, but since there are so many of them it does not make sense to label all of them. We might for instance label the proteins that have a 2logfold change above 2, which means they are overexpressed by 400%.

Lets have a look at hsc to cmp. We see that three protein groups are significantly overexpressed above 400%. P68033, P05213 and Q9WUA1. We can look up their Uniprot acession ID's and we find the following information:

P68033 is an actin, actins are highly conserved proteins that are involved in various types of cell motility and are ubiquitously expressed in all eukaryotic cells.

P05213 is tubulin, tubulin is the major constituent of microtubules, a cylinder consisting of laterally associated linear protofilaments composed of alpha- and beta-tubulin heterodimers.

Q9WUA1 is an inhibitory factor that binds to WNT proteins and inhibits their activities. May be involved in mesoderm segmentation.

Magda tried changing the point size of the legend but in vain...

```{r}
#This is a toyplot to try out new things
df_volcano_clp_hsc|>
  mutate(label = case_when(expression == "overexpressed" & fold_log2 > 2 ~ protein_groups,
                           .default = "")) |> 
  ggplot(aes(x = fold_log2, y = desc(log10(q_val)), color = expression))+
  geom_point(size=0.5, alpha = 0.5)+
  theme_classic()+
  geom_text_repel(aes(label = label), size = 3)+
  labs(title = "Expression of GMP relative to CMP",
       x = "Log2fold-change",
       y = "-log10(q)",
       caption = "proteomics data")+
  theme(legend.position = "bottom", legend.title = element_blank())+
  guides(color = guide_legend(overrisde.aes = list(size = 1000)))+
  #The horizontal line that indicates significance - remember to check if it correct with qvalue and all
#  geom_hline(yintercept = 2)+
  scale_color_manual(values = c("overexpressed" = "#D55E00", "underexpressed" = "#0072B2", "not_significant" = "gray"))
  
```

```{r}
hsc_cmp_plot + hsc_clp_plot + cmp_gmp_plot + cmp_mep_plot

hsc_cmp_plot
```
